src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.cs

```diff
@@ -198,11 +198,41 @@ public int BytesTransferred
 
         public event EventHandler<SocketAsyncEventArgs>? Completed;
 
+        private void OnCompletedInternal(SocketAsyncEventArgs e)
+        {
+            if (SocketsTelemetry.Log.IsEnabled())
+            {
+                e.AfterConnectAcceptTelemetry();
+            }
+
+            OnCompleted(e);
+        }
+
         protected virtual void OnCompleted(SocketAsyncEventArgs e)
         {
             Completed?.Invoke(e._currentSocket, e);
         }
 
+        private void AfterConnectAcceptTelemetry()
+        {
+            if (SocketsTelemetry.Log.IsEnabled())
```

Comment:
All of the call sites are checking this.  Why check again immediately?

Reply:
Is the linker capable of removing the method entirely if all call sites are guarded already?
If so, this isn't needed as it's redundant.

Reply:
It should be; if it's not I expect we have much bigger problems.  @eerhardt?

Reply:
When `EventSource.IsSupported` is disabled in the project, `IsEnabled()` is re-written to return false:

https://github.com/dotnet/runtime/blob/786548f0c1b76eacb64dc37fa3db8a5f2d68fbc0/src/libraries/System.Private.CoreLib/src/ILLink/ILLink.Substitutions.Shared.xml#L3-L8

When a method returns a constant value, the ILLinker can eliminate branches that are based off the return of that method. So in this case, if `EventSource.IsSupported` is disabled, `AfterConnectAcceptTelemetry()` will be completely removed.

Reply:
Removed the `IsEnabled` check from the method itself.